src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1457,6 +1461,68 @@ public void setConcurrentCompactors(int value)
         CompactionManager.instance.setConcurrentCompactors(value);
     }
 
+    public String getAuthenticator()
+    {
+        return DatabaseDescriptor.getAuthenticator().getClass().getName();
+    }
+
+    public void setAuthenticator(String value)
+    {
+        if (value == null)
+            throw new IllegalArgumentException("Authenticator cannot be null");
+
+        try {
+            Class<?> cl = Class.forName(value);
+
+            // Ensure the class implements IAuthenticator
+            if (IAuthenticator.class.isAssignableFrom(cl)) {
```

Comment:
once 19946 is in, this should be done by means of ParameterizedClass, this logic is redundant, imho.